File: find_the_highest_altitude.cpp
class Solution {
    public:
   int largestAltitude(vector<int>& gain) {
     // Your code goes here
    int altitude = 0;
    int highest = 0;
    
    for (int netGain : gain) {
        altitude += netGain;
        highest = std::max(highest, altitude);
    }
    
    return highest;
   }
};

File: find_the_highest_altitude.js
/**
 * @param {number[]} gain
 * @return {number}
 */
function largestAltitude(gain) {
    let altitude =0;
    let highest = 0;
    gain.forEach((g)=>{
        altitude+=g;
        highest = Math.max(highest,altitude);
    })
    return highest;
};